### PR TITLE
Finalize windows support

### PR DESCRIPTION
**Pull request by hasufell**

### CODE REVIEW

File: src/Codec/Archive/Internal/Pack/Common.hs [hunk 1]
```diff
@@ -23,4 +23,4 @@ mkEntry :: FilePath -> IO (Entry FilePath BS.ByteString)
 mkEntry fp = do
     status <- getFileStatus fp
     content' <- mkContent fp status
-    pure $ Entry fp content' (fileMode status) (Ownership Nothing Nothing (fromIntegral $ fileOwner status) (fromIntegral $ fileGroup status)) Nothing
+    pure $ Entry fp content' (fromIntegral (fileMode status)) (Ownership Nothing Nothing (fromIntegral $ fileOwner status) (fromIntegral $ fileGroup status)) Nothing
```
Comment (hasufell) [inline src/Codec/Archive/Internal/Pack/Common.hs:26]: Can't judge the safety of this... `cabal test` succeeds.

File: gen-c-files.sh [hunk 2]
```diff
@@ -74,8 +74,18 @@ case "$(uname -s)" in
 			-e '/define HAVE_SYS_ACL_H/d' \
 			-e '/define HAVE_EXT2FS_EXT2_FS_H/d' \
 			config.h > ../c/autoconf-freebsd/config.h ;;
-
-	# TODO: windows
+	MSYS*|MINGW*)
```
Comment (hasufell) [inline gen-c-files.sh:77]: Tested, works nicely with bundled too.

File: libarchive.cabal [hunk 3]
```diff
@@ -83,6 +84,9 @@ library
         unix-compat >=0.1.2.1,
         deepseq >=1.4.0.0
 
+    if os(windows)
+        cc-options: -U__GCC_ASM_FLAG_OUTPUTS__
```
Comment (hasufell) [inline libarchive.cabal:88]: This is the workaround mentioned in https://github.com/haskell/c2hs/issues/199#issuecomment-649476689

File: src/Codec/Archive/Internal/Pack/Common.hs (hunk 1 above)
Comment (hasufell) [reply src/Codec/Archive/Internal/Pack/Common.hs:26]: I was lying, they don't:

```
$ cabal run -w ghc-9.0.1 -f-system-libarchive libarchive-test
Build profile: -w ghc-9.0.1 -O1
In order, the following will be built (use -v for more details):
 - libarchive-3.0.2.0 (test:libarchive-test) (additional components to build)
Preprocessing library for libarchive-3.0.2.0..
Building library for libarchive-3.0.2.0..
Preprocessing test suite 'libarchive-test' for libarchive-3.0.2.0..
Building test suite 'libarchive-test' for libarchive-3.0.2.0..

roundtrip
  sucessfully unpacks/packs (test/data\aarch64-linux-dist.tar)
  should repack (zip/test/data\aarch64-linux-dist.tar)
  should repack (cpio)
  should repack (shar)
  should repack (7zip)
  works on nonstandard bytestring (test/data\aarch64-linux-dist.tar)
  works on strict bytestring (test/data\aarch64-linux-dist.tar)
  works on strict bytestring (test/data\aarch64-linux-dist.tar)
  unpacks test/data\aarch64-linux-dist.tar
  reads test/data\aarch64-linux-dist.tar FAILED [1]

Failures:

  test\Spec.cpphs:45:30:
  1) roundtrip reads test/data\aarch64-linux-dist.tar
       predicate failed on: Left ArchiveFatal

  To rerun use: --match "/roundtrip/reads test/data\\aarch64-linux-dist.tar/"

Randomized with seed 1239997964

Finished in 0.1761 seconds
10 examples, 1 failure
```

Comment (hasufell) [reply src/Codec/Archive/Internal/Pack/Common.hs:26]: Test failures are unreliable... I think this has to do with file locking.

File: test/Spec.cpphs [hunk 4]
```diff
@@ -107,3 +116,9 @@ main = do
                 [ stripOwnership (simpleFile "a.txt" (NormalFile "text")) ]
             xcontext "having entry without timestamp" . itPacksUnpacks $
                 [ stripTime (simpleFile "a.txt" (NormalFile "text")) ]
+  where
+#ifdef mingw32_HOST_OS
```
Comment (hasufell) [inline test/Spec.cpphs:120]: parallel tests fail due to file locking on windows

**Review by vmchale — APPROVED** [review]

**Comment by hasufell** [comment]
@vmchale so is the `fromIntegral` safe there?

**Pull request merged**